sdk/resourcemanager/Azure.ResourceManager.Core/tests/Scenario/TagsOperationsTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Core.Tests
+{
+    public class TagsOperationsTests : ResourceManagerTestBase
+    {
+        private ResourceGroup _rg;
+        private string _rgPrefix = "rg";
+
+        public TagsOperationsTests(bool isAsync)
+            : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+
+        [SetUp]
+        public async Task SetUpAsync()
+        {
+            _rg = await Client.DefaultSubscription.GetResourceGroups().Construct(LocationData.WestUS2).CreateOrUpdateAsync(Recording.GenerateAssetName(_rgPrefix));
+            _rg = await _rg.AddTagAsync("key1", "value1");
+            _rg = await _rg.AddTagAsync("key2", "value2");
+        }
+
+        [TestCaseSource(nameof(OperationType))]
+        [RecordedTest]
+        public async Task GetTagsRestOperations(TagsPatchResourceOperation type, IDictionary<string, string> expect)
+        {
+            var operation = Client.DefaultSubscription.GetTagsOperations();
```

Comment:
we don't need to test the rest operations, what we should do instead is
Get a resourceGroup object
call rg.AddTag
call rg.Get
validate the tags contain the expected result

we should do this for various scenarios like, try to add a tag that already exists, try to add a tag that already exists with the same key and value, try to add a tag that doesn't exist etc

we should do this for all the publicly exposed "tag" methods on resources
AddTag
RemoveTag
SetTags

as well as their "start" variants

Reply:
Done